src/mscorlib/src/System/Runtime/Intrinsics/X86/Avx2.cs

```diff
@@ -544,243 +544,243 @@ public static class Avx2
         /// __m128i _mm_i32gather_epi32 (int const* base_addr, __m128i vindex, const int scale)
         ///   VPGATHERDD xmm, vm32x, xmm
         /// </summary>
-        public static unsafe Vector128<int> GatherVector128(int* baseAddress, Vector128<int> index, byte scale) => GatherVector128(baseAddress, index, scale);
+        public static unsafe Vector128<int> GatherVector128WithVector128Int32Indices(int* baseAddress, Vector128<int> index, byte scale) => GatherVector128WithVector128Int32Indices(baseAddress, index, scale);
```

Comment:
While I can imagine it being easier for JIT implementation, these names are a bit unwieldy from a user's point of view IMO.

Reply:
Yes, but if we only rely on overload to distinguish these intrinsics, the current named-intrinsic framework of RyuJIT has to be changed. Meanwhile, in the future, other platforms (e.g., .NET Native, Mono, etc) are also hard to port hardware intrinsic.

JIT experts may have some thoughts? @CarolEidt @mikedn @AndyAyersMS

Reply:
The JIT intrinsic plumbing needs to be able to deal with overloads. It does it today in several cases.  For example, `CORINFO_INTRINSIC_Abs` can be either `Abs(float)` or `Abs(double)`.  Or there are multiple intrinsic overloads of `Vector::.ctor`. If the named intrinsics are meant to be the one true way to identify intrinsics in future, it should be just fixed there.

> Meanwhile, in the future, other platforms (e.g., .NET Native, Mono, etc) are also hard to port hardware intrinsic.

I do not think so.

Reply:
It doesn't seem to me that it will be all that difficult to distinguish these, and given that it is not a common case, we could either choose to simply generate different `NI*` intrinsic names in the importer, based on the type of the relevant argument, **or** we could postpone that analysis until codegen, and look at the base type of the SIMD argument to determine which instruction to generate.

Reply:
And I would say that, in general, the API shape is more important than the complexity of the JIT implementation, as long as it is both "pay for play" (i.e. the complexity is only incurred for the case in question) and the complexity is not very high.

Reply:
Thanks, will retain the current APIs and address in JIT.